pkg/odo/cli/service/ui/ui_test.go

```diff
@@ -0,0 +1,269 @@
+
+		categories := getServiceClassesCategories(classes)
+		expected := []string{"footag", "other"}
+		if !reflect.DeepEqual(expected, categories) {
+			t.Errorf("test failed, expected %v, got %v", expected, categories)
+		}
+	})
+}
+
+func TestGetServicePlanNames(t *testing.T) {
+	t.Run("GetServicePlanNames should work", func(t *testing.T) {
+		foo := testingutil.FakeClusterServicePlan("foo", 1)
+		bar := testingutil.FakeClusterServicePlan("bar", 2)
+		boo := testingutil.FakeClusterServicePlan("boo", 3)
+
+		plans := GetServicePlanNames(map[string]beta1.ClusterServicePlan{"foo": foo, "bar": bar, "boo": boo})
+		expected := []string{"bar", "boo", "foo"}
+		if !reflect.DeepEqual(expected, plans) {
+			t.Errorf("test failed, expected %v, got %v", expected, plans)
+		}
+	})
+}
+
+func TestWrapIfNeeded(t *testing.T) {
+	tests := []struct {
+		name       string
+		input      string
+		prefixSize int
+		expected   string
+	}{
+		{
+			name:       "empty string, empty prefix",
+			input:      "",
+			prefixSize: 0,
+			expected:   "",
+		},
+		{
+			name:       "short string, empty prefix should not be wrapped",
+			input:      "foo bar baz",
+			prefixSize: 0,
+			expected:   "foo bar baz",
+		},
+		{
+			name:       "short string, empty prefix should not be wrapped with default width",
+			input:      "foo bar baz",
+			prefixSize: 2,
+			expected:   "foo bar baz",
+		},
+		{
+			name:       "short string, long prefix should wrap",
+			input:      "foo bar baz",
+			prefixSize: 78,
+			expected:   "foo\nbar\nbaz",
+		},
+		{
+			name:       "long string, empty prefix should wrap",
+			input:      "0123456789 0123456789 0123456789 0123456789 0123456789 0123456789 0123456789 0123456789",
+			prefixSize: 0,
+			expected:   "0123456789 0123456789 0123456789 0123456789 0123456789 0123456789 0123456789\n0123456789",
+		},
+		{
+			name:       "long string, short prefix should wrap",
+			input:      "0123456789 0123456789 0123456789 0123456789 0123456789 0123456789 0123456789 0123456789",
+			prefixSize: 2,
+			expected:   "0123456789 0123456789 0123456789 0123456789 0123456789 0123456789 0123456789\n0123456789",
+		},
+		{
+			name:       "long string, longer prefix should wrap more",
+			input:      "0123456789 0123456789 0123456789 0123456789 0123456789 0123456789 0123456789 0123456789",
+			prefixSize: 10,
+			expected:   "0123456789 0123456789 0123456789 0123456789 0123456789 0123456789\n0123456789 0123456789",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			output := wrapIfNeeded(tt.input, tt.prefixSize)
+			if tt.expected != output {
+				t.Errorf("test failed, expected %s, got %s", tt.expected, output)
+			}
+		})
+	}
+}
+
+func TestEnterServicePropertiesInteractively(t *testing.T) {
+	// TODO: this test is currently skipped because it is not currently working properly. :(
+	t.Skip("TODO: Skip this test until we can figure out what is wrong with it")
```

Comment:
+1, Would be happy to see this in a separate issue :)

Reply:
Done: #1206